### PR TITLE
Fix bug where radio.select passes the previous value to the function instead of the selected value

### LINKED ISSUE #6216: Radio select gives the previous value not the current value
### Describe the bug

When using `radio.select`, the value passed to the backend is the previous value not the value that has just been selected. This is different to how dropdown select works for example.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr


with gr.Blocks() as demo:
    with gr.Row():
        with gr.Column():
            dd1 = gr.Radio(choices=["a", "b", "c"])
        with gr.Column():
            dd2 = gr.Radio(choices=["a", "b", "c"])
        dd1.select(lambda s: s, dd1, dd2)

demo.launch()
```

### Screenshot

radio_change

### Logs

_No response_

### System Info

```shell
Gradio Environment Information:
------------------------------
Operating System: Darwin
gradio version: 4.0.2
gradio_client version: 0.7.0

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
altair: 4.2.0
fastapi: 0.101.0
ffmpy: 0.3.1
gradio-client==0.7.0 is not installed.
httpx: 0.23.0
huggingface-hub: 0.18.0
importlib-resources: 6.0.1
jinja2: 3.1.2
markupsafe: 2.1.1
matplotlib: 3.7.2
numpy: 1.26.0
orjson: 3.9.4
packaging: 22.0
pandas: 1.5.3
pillow: 9.2.0
pydantic: 2.4.0
pydub: 0.25.1
python-multipart: 0.0.6
pyyaml: 6.0
requests: 2.28.1
semantic-version: 2.10.0
tomlkit==0.12.0 is not installed.
typer: 0.9.0
typing-extensions: 4.8.0
uvicorn: 0.23.2
websockets: 11.0.3
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2023.6.0
httpx: 0.23.0
huggingface-hub: 0.18.0
packaging: 22.0
requests: 2.28.1
typing-extensions: 4.8.0
websockets: 11.0.3
```

### Severity

I can work around it

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

Fixes #6216 

```python
import gradio as gr


with gr.Blocks() as demo:
    with gr.Row():
        with gr.Column():
            dd1 = gr.Radio(choices=["a", "b", "c"])
        with gr.Column():
            dd2 = gr.Radio(choices=["a", "b", "c"])
        dd1.select(lambda s: s, dd1, dd2)

demo.launch()
```

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: js/radio/shared/Radio.svelte [hunk 1]
```diff
@@ -10,8 +10,19 @@
 	export let selected: string | null = null;
 
 	const dispatch = createEventDispatcher<{ input: string | number }>();
+	let is_selected = false;
```
Comment (pngwn) [inline js/radio/shared/Radio.svelte:13]: Is this used anywhere? Looks like we can bring it into the function completely.

**Review by pngwn — APPROVED** [review]
Looks great other than that small nit. Thanks @freddyaboulton!

File: js/radio/shared/Radio.svelte (hunk 1 above)
Comment (freddyaboulton) [reply js/radio/shared/Radio.svelte:13]: To apply css for the selected class!

Comment (pngwn) [reply js/radio/shared/Radio.svelte:13]: ah cool

**Pull request merged**